docs/source/traffic_assignment.rst

```diff
@@ -0,0 +1,554 @@
+  the technical detail), and it is set to 1 by default.  If the **pce** for a
+  certain class should be different than one, one can make a quick method call.
+
+Example:
+
+::
+
+  tc = TrafficClass(graph_car, matrix_car)
+
+  tc2 = TrafficClass(graph_truck, matrix_truck)
+  tc2.set_pce(1.9)
+
+
+To add traffic classes to the assignment instance it is just a matter of making
+a method call:
+
+::
+
+  assig.set_classes([tc, tc2])
+
+
+setting VDF Parameters
+~~~~~~~~~~~~~~~~~~~~~~
+
+Parameters for VDF functions can be passed as a fixed value to use for all
+links, or as graph fields. As it is the case for the travel time and capacity
+fields, VDF parameters need to be consistent across all graphs.
+
+Because AequilibraE supports different parameters for each link, its
+implementation is the most general possible while still preserving the desired
+properties for multi-class assignment, but the user needs to provide individual
+values for each link **OR** a single value for the entire network.
+
+Setting the VDF parameters should be done **AFTER** setting the VDF function of
+choice and adding traffic classes to the assignment, or it will **fail**.
+
+To choose a field that exists in the graph, we just pass the parameters as
+follows:
+
+::
+
+  assig.set_vdf_parameters({"alpha": "alphas", "beta": "betas"})
+
+
+To pass global values, it is simply a matter of doing the following:
+
+::
+
+  assig.set_vdf_parameters({"alpha": 0.15, "beta": 4})
+
+
+Setting final parameters
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are still three parameters missing for the assignment.
+
+* Capacity field
+
+* Travel time field
+
+* Equilibrium algorithm to use
+
+::
+
+  assig.set_capacity_field("capacity")
+  assig.set_time_field("free_flow_time")
+  assig.set_algorithm(algorithm)
+
+Finally, one can execute assignment:
+
+::
+
+  assig.execute()
+
+:ref:`convergence_criteria` is discussed below.
+
+Multi-class Equilibrium assignment
+----------------------------------
+
+By introducing equilibrium assignment [1] with as many algorithms as we have, it
+makes sense to also introduce multi-class assignment, adding to the pre-existing
+capability of assigning multiple user-classes at once.  However, multi-class
+equilibrium assignments have strict technical requirements and different
+equilibrium algorithms have slightly different resource requirements.
+
+Cost function
+~~~~~~~~~~~~~
+
+It is currently not possible to use custom cost functions for assignment, and
+the only cost function available to be minimized is simply travel time.
+
+.. _technical_requirements_multi_class:
+
+Technical requirements
+~~~~~~~~~~~~~~~~~~~~~~
+
+This documentation is not intended to discuss in detail the mathematical
+requirements of multi-class traffic assignment, which can be found discussed in
+detail on `Zill et all. <https://doi.org/10.1177%2F0361198119837496>`_
+
+A few requirements, however, need to be made clear.
```

Comment:
I think we should try to make clear that these are not requirements for multi-class user equilibrium in general, but only to express it as a constrained convex optimisation problem, which in turn is much faster to solve than the generic variational inequality problem.

Reply:
other than that this is quite extensive and good documentation

Reply:
They are requirements given the solver we are using. are they?

Reply:
But you have a point. Do you want to have a crack at rephrasing it?

Reply:
yes, they are requirements for our method

Reply:
I think I would just add a sentence after "A few requirements...", reading something like: For the methods implemented in Aequilibrae, we rely on the existence of a constrained convex optimization formulation of the multi-class user equilibrium problem, which in turn is based on the following assumptions.

Reply:
I rephrased it